pkgs/uefi-firmware/edk2-uefi-dtb.patch

```diff
@@ -1,15 +1,14 @@
 diff --git a/Silicon/NVIDIA/Library/DxeDtPlatformDtbLoaderLib/DxeDtPlatformDtbKernelLoaderLib.c b/Silicon/NVIDIA/Library/DxeDtPlatformDtbLoaderLib/DxeDtPlatformDtbKernelLoaderLib.c
-index 1defbed..d16b86e 100644
+index ba009757..71593409 100644
 --- a/Silicon/NVIDIA/Library/DxeDtPlatformDtbLoaderLib/DxeDtPlatformDtbKernelLoaderLib.c
 +++ b/Silicon/NVIDIA/Library/DxeDtPlatformDtbLoaderLib/DxeDtPlatformDtbKernelLoaderLib.c
-@@ -705,9 +705,7 @@ DtPlatformLoadDtb (
-   Status = gBS->CreateEventEx (
-                   EVT_NOTIFY_SIGNAL,
-                   TPL_CALLBACK,
--                  UefiDtbBoot ?
--                  UpdateFdt :
--                  InstallFdt,
-+                  UpdateFdt,
-                   NULL,
-                   &gEfiEventReadyToBootGuid,
-                   &ReadyToBootEvent
+@@ -561,8 +561,7 @@ InstallFdt (
+   }
+ 
+   PlatformType = TegraGetPlatform ();
+-  if ((PlatformType != TEGRA_PLATFORM_SILICON) ||
+-      (PlatformResourceInfo->BootType == TegrablBootRcm))
++  if (TRUE) // Always use DTB from UEFI

```

Comment:
Why was this change made? Does this prevent the [`hardware.deviceTree`](https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=hardware.deviceTree) options from working? We use it to merge the kernel-supplied device tree with camera overlays.

Reply:
The original patch no longer applied since upstream refactored this area of the codebase.  Unlike upstream's edk2 builds, we don't use default to using the `kernel-dtb` partitions to load/store the device tree for the Linux kernel, but instead always re-use the device tree included with the uefi firmware.  The change I made here shouldn't change behavior from how it worked previously.  `hardware.deviceTree` options should still work.